networkx/algorithms/coloring/greedy_coloring.py

```diff
@@ -65,14 +65,47 @@ def strategy_smallest_last(G, colors):
 
     """
     H = G.copy(with_data=False)
-
-    def min_and_pop():
-        """Removes and returns the minimum degree node in ``H``."""
-        node = min(H, key=H.degree)
-        H.remove_node(node)
-        return node
-
-    return reversed([min_and_pop() for x in G])
+    result = []
+
+    # Build initial degree list (i.e. the bucket queue data structure)
+    degrees = defaultdict(set)  # set(), for fast random-access removals
+    for node, d in H.degree():
+        degrees[d].add(node)
+        try:
+            H.lbound = min(H.lbound, d)  # Lower bound on min-degree.
```

Comment:
Fewer lines of code, exception checking is slow, and the exception occurs exactly once at the first iteration.